output-html.inc

```diff
@@ -1,5 +1,40 @@
 <?php
 
+function check_for_gpg_fingerprint_on_mozilla_keyserver($gpg_key) {
+  $return_bool = false;
+  if (is_callable('curl_init')) {
+    $ch = curl_init();
+    $search_url = 'http://gpg.mozilla.org/pks/lookup?search=0x' . $gpg_key . '&op=vindex';
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+    curl_setopt($ch,CURLOPT_URL,$search_url);
+    $output=curl_exec($ch);
+    $return_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    if (isset($return_code)) {
+      if ($return_code == "404"){
+        $return_bool = false;
+      } elseif ($return_code == "200"){
+        $return_bool = true;
+      }
+    }
+    curl_close($ch);
+  }
+  return $return_bool;
+
+}
+
+function output_pgpFingerprint($s) {
+  $stripped_fingerprint = str_replace(" ", "", $s);
+  $show_link = check_for_gpg_fingerprint_on_mozilla_keyserver($stripped_fingerprint);
```

Comment:
1) This strikes me as excessively complicated. What would happen if we just showed the link and that link was a 404? Nothing, really. People would click and see a 404 because the keyserver does not have the key. Too bad. If you care a lot, then I suggest on login.mozilla.org you either write a message urging people to submit to our keyserver OR you reject the user's submission UNLESS it can be found on the server.

2) Unless I am mistaken, this will be called for *every* vcard on a search results page. If you enter something like "Toronto" and get 100 results with PGP keys, this will push 100 CURL requests to the gpg server, which will at least be slow.

I'd say, don't bother checking if we can link to it, just assume we can. You *could* put a title attribute on the link saying something like "search for this key on the mozilla key server", implying it's a *search* and searches can return nothing if you're unlucky.